app/src/test/java/ar/com/wolox/android/example/ui/example/LoginPresenterTest.kt

```diff
@@ -0,0 +1,155 @@
+import ar.com.wolox.android.example.utils.UserSession
+import ar.com.wolox.wolmo.core.tests.CoroutineTestRule
+import ar.com.wolox.wolmo.core.tests.WolmoPresenterTest
+import ar.com.wolox.wolmo.networking.retrofit.handler.NetworkResponse
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import okhttp3.Headers
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito
+import retrofit2.Response
+
+@ExperimentalCoroutinesApi
+class LoginPresenterTest : WolmoPresenterTest<LoginView, LoginPresenter>() {
+
+    @get:Rule
+    val coroutineTestRule = CoroutineTestRule(runOnAllTests = true)
+
+    @Mock
+    lateinit var userSession: UserSession
+
+    @Mock
+    lateinit var userRepository: UserRepository
+
+    override fun getPresenterInstance() = LoginPresenter(userSession, userRepository)
+
+    @Test
+    fun UserAndPaswordEmptyValidation() {
+        // Usuario y contraseña vacio
+        val username = ""
+        val password = ""
+
+        presenter.onLoginButtonClicked(username, password)
+
+        verify(view, times(1)).showError(Extras.UserLogin.USERNAME)
+        verify(view, times(1)).showError(Extras.UserLogin.PASSWORD)
+    }
+
+    @Test
+    fun UserIcorrectAndPaswordEmptyValidation() {
+        // Usuario invalido y contraseña vacia
+        val username = "prueba"
+        val password = ""
+
+        presenter.onLoginButtonClicked(username, password)
+
+        verify(view, times(1)).showError(Extras.UserLogin.VALID_EMAIL)
+        verify(view, times(1)).showError(Extras.UserLogin.PASSWORD)
+    }
+
+    @Test
+    fun UserCorrectAndPaswordEmtpyValidation() {
+        // Usuario correcto y contraseña vacia
+        val username = "prueba@prueba.com"
+        val password = ""
+
+        presenter.onLoginButtonClicked(username, password)
+
+        verify(view, times(1)).showError(Extras.UserLogin.PASSWORD)
+    }
+
+    @Test
+    fun requestErrorLogin() = runBlocking {
+
+        // Usuario y contraseña incorrectas
+        val username = "prueba@prueba.com"
+        val password = "123"
+
+        // Usuario response
+        val authenticationbody = AuthenticationBody(username, password)
+
+        val response = Mockito.mock(Response::class.java) as Response<User>
+        whenever(response.code()).doReturn(401)
+        whenever(userRepository.signIn(authenticationbody)).doReturn(NetworkResponse.Error(response))
+
+        presenter.loginNetwork(username, password).join()
+
+        verify(view, times(1)).showError(Extras.Constantes.ERROR_USER_PASS)
+    }
+
+    @Test
+    fun requestFailedLogin() = runBlocking {
+
+        // Usuario y contraseña correctas pero hay error de conexion a la red
+        val username = "melvin.lambert15@example.com"
```

Comment:
As a suggestion, you can move all these hardcoded values to a test constants file.